DOC: read-through and edit the new jax tutorials (#20306)
jax-ml/jax: docs/tutorials/debugging.md

  z = x / y
  breakpoint_if_nonfinite(z)
  return z

f(2., 1.) # ==> No breakpoint
```

```{code-cell}
:tags: [skip-execution]

f(2., 0.) # ==> Pauses during execution
```

## JAX `debug.callback` for more control during debugging

> Both {func}`jax.debug.print` and {func}`jax.debug.breakpoint` are implemented in terms of

Reviewer: ```suggestion
Both {func}`jax.debug.print` and {func}`jax.debug.breakpoint` are implemented using
```

Author: Done

the more flexible {func}`jax.debug.callback`, which gives greater control over the

> host-side logic executed during the callback.

Reviewer: ```suggestion
host-side logic executed via a Python callback.
```
"the callback" is confusing because you haven't introduced any callback yet I think?

Author: Done
